### PR TITLE
fix: add correct handling for conversation delete (AR-2298)

### PR DESCRIPTION
**Pull request by yamilmedina**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

Currently, when we are getting the conversation delete event, can be already handled in background, or during the sync process, so whenever when we are trying to delete this conversation the flow was not terminating creating an infinite wait state.

### Solutions

Provide a query, repo level, that allow us to get a Nullable conversation and with that, handle it in the correct way in the `ConversationEventReceiver`

#### Test Coverage (Optional)

- [x] I have added automated test to this contribution

#### How to Test

The STR are in the current ticket https://wearezeta.atlassian.net/browse/AR-2298

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [x] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/ConversationRepository.kt [hunk 1]
```diff
@@ -400,6 +402,13 @@ class ConversationDataSource(
             .map(conversationMapper::fromDaoModel)
             .wrapStorageRequest()
 
+    override suspend fun getConversationById(conversationId: ConversationId): Conversation? =
+        conversationDAO.observeGetConversationByQualifiedID(idMapper.toDaoModel(conversationId))
+            .map {
+                if (it != null) conversationMapper.fromDaoModel(it)
```
Comment (Garzas) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/ConversationRepository.kt:408]: Maybe use ?.let {}

**Review by Garzas — APPROVED** [review]
Just small suggestion, LGTM 🚀

File: logic/src/commonTest/kotlin/com/wire/kalium/logic/data/conversation/ConversationRepositoryTest.kt [hunk 2]
```diff
@@ -1372,6 +1268,24 @@ class ConversationRepositoryTest {
...
+    }
+
+    @Test
+    fun givenAConversationId_WhenTheConversationExists_ShouldReturnAConversationInstance() = runTest {
+        val conversationId = ConversationId("conv_id", "conv_domain")
+        val (_, conversationRepository) = Arrangement().withExpectedConversation(TestConversation.ENTITY).arrange()
+
+        val result = conversationRepository.getConversationById(conversationId)
+        assertNotNull(result)
+    }
```
Comment (vitorhugods) [inline logic/src/commonTest/kotlin/com/wire/kalium/logic/data/conversation/ConversationRepositoryTest.kt:1287]: 👨🏼‍🍳🤌🏼

File: logic/src/commonTest/kotlin/com/wire/kalium/logic/sync/receiver/ConversationEventReceiverTest.kt [hunk 3]
```diff
@@ -285,12 +285,32 @@ class ConversationEventReceiverTest {
...
+
+        eventReceiver.onEvent(event)
+
+        with(arrangement) {
+            verify(conversationRepository)
+                .suspendFunction(conversationRepository::deleteConversation)
+                .with(eq(TestConversation.ID))
+                .wasNotInvoked()
+        }
+    }
```
Comment (vitorhugods) [inline logic/src/commonTest/kotlin/com/wire/kalium/logic/sync/receiver/ConversationEventReceiverTest.kt:306]: 👨🏼‍🍳🤌🏼

Awesome

**Review by gongracr — APPROVED** [review]
Looking great. All these coupling between suspended discrete functions and flows can get hide a lot of future pain when not implemented in the right way. Great catch anyways and kudos for the extra tests! 🪱 🦅

**Pull request merged**